### ISSUE TITLE
[HOLD for payment 2022-12-27] [$2000] Android - Long-press to edit a comment > keyboard opens and is immediately dismissed causing a jarring animation

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#11048]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Launch the app
2. Log in with expensifail account
3. Select any user
4. Send first message
5. Hold the finger on you message and select Edit

## Expected Result:
Edit comment transaction is smooth, app is not jumping

## Actual Result:
Edit comment transaction is not smooth, app is jumping

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Android

**Version Number:** 1.2.1.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/190658616-ecbf0884-a1dd-45be-ab9e-26ae726456e9.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by arielgreen** [comment]
Asked for an assist reproducing this [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1666294160801929).

**Comment by arielgreen** [comment]
Thanks @mananjadhav!

**Comment by arielgreen** [comment]
https://www.upwork.com/jobs/~019bba3822889fb13a

**Comment by arielgreen** [comment]
Awaiting proposal

**Comment by arielgreen** [comment]
Price increased.

**Comment by thesahindia** [comment]
@arielgreen, let's change this to weekly

**Comment by arielgreen** [comment]
@thesahindia we are now keeping E/App issues as daily

**Comment by arielgreen** [comment]
price increased

**Comment by AndrewGable** [comment]
Still waiting on proposals

**Comment by arielgreen** [comment]
Upped price!

**Comment by arielgreen** [comment]
Awaiting proposals.

**Comment by JmillsExpensify** [comment]
@AndrewGable To help clear out the extensive backlog of /App bugs, we're putting the spotlight on all bugs older than 4 weeks old. To help unblock the roadmap and get our bug pipeline back in equilibrium, can you:

 - Decide whether any proposals currently meet our guidelines and can be approved as-is
 - For any that can't, please take this issue internal and treat it as one of your highest priorities
 - If you have any questions, don't hesitate to start a discussion in #bug-zero

Thanks everyone!

**Comment by AndrewGable** [comment]
No proposals exist, I can look into it.

**Comment by AndrewGable** [comment]
First I am trying to understand if there is really any "jumping" or if this is just a place where better animations should exist...

**Comment by AndrewGable** [comment]
Good question @arielgreen - I have been trying to understand the bug a bit more before writing code. One difference I've seen is that on iOS we always show the keyboard, sometimes on Android we are not... Not sure if that's exactly the "bug" but it's one difference I noticed.

**Comment by JmillsExpensify** [comment]
I kind of think we still have a lot to learn with Keyboard. A part of me thinks we need to step back and take a react navigation type approach, where we look at the behaviors more holistically and figure out how we want the app to behavior/perform across all platforms.

**Comment by AndrewGable** [comment]
Going to look at this today with @JmillsExpensify comment in mind.

**Comment by AndrewGable** [comment]
Going to re-test since keyboard avoiding view is done being refactored

**Comment by AndrewGable** [comment]
I was hoping to lump it together with https://github.com/Expensify/App/issues/9252 and hopefully have @hannojg look into it.

**Comment by JmillsExpensify** [comment]
Oh thanks! That jogs my memory, I'll float in our room then ad remove the external label.

**Comment by JmillsExpensify** [comment]
Floated with Margelo. They expect to look into this soon.

**Comment by tgolen** [comment]
I'm going to take a look into this today and see if I can determine anything. Maybe it will help the investigation move forward

**Comment by AndrewGable** [comment]
Thanks @tgolen!

**Comment by tgolen** [comment]
I was having some inconsistent results trying to reproduce this so I have [reached out to QA](https://expensify.slack.com/archives/C9YU7BX5M/p1670957269489749) to get help reproducing it on physical devices.

**Comment by tgolen** [comment]
OK, I think I found the culprit, though, I don't have a complete solution yet. I'd love to get others thoughts on this.

Problem: When the edit button is clicked in the context menu, there is a small animation to display the comment composer (it slides up into view). It is only maybe 100ms and you can see it in this video:


https://user-images.githubusercontent.com/1228807/207434163-a53e68ec-e412-4672-99cd-90af71c40bc4.mp4

We are attempting to put focus into that textInput while it's still animating, and that causes a couple problems:

1. A race condition where the input isn't mounted yet because the animation is still happening and therefore it never gets focus and the keyboard never opens
2. Janky animations (not sure the cause of this)

The only solution I have for now is to wrap this code in a `setTimeout()` with 100ms delay:

https://github.com/Expensify/App/blob/f036c7db8fe243d53c88d393699dce6a34776afc/src/pages/home/report/ReportActionItem.js#L98

I was not able to determine where or why that animation happens... if someone knows, can you help point me to it? That could help us get to the root cause and prevent the need for a `setTimeout()` to "guess" when the animation is done.

cc @marcaaron @luacmartins @Julesssss @jasperhuangg @roryabraham 

We could use something like hooking into `reanimated` to better detect when the animation is done maybe? 🤷

**Comment by marcaaron** [comment]
I am not entirely sure if there is an animation we can control happening. IIRC the "edit comment" replaces the report action item itself so you could actually be seeing that swap happening with whatever native animation gets applied when items are added and removed from a `FlatList`.

Separately, I would check to make sure that this `setTimeout()` solution doesn't break the focus behavior for iOS Safari. While researching another issue we discovered that [text input focus will break on iOS Safari unless it is hooked into a user interaction](https://gist.github.com/cathyxz/73739c1bdea7d7011abb236541dc9aaa) (e.g. triggered via a "press" event and not in a `setTimeout()` callback).

**Comment by marcaaron** [comment]
https://github.com/Expensify/App/blob/ab803314adf833353a9465f018255ab935f425e0/src/pages/home/report/ReportActionItem.js#L148-L164

**Comment by chrispader** [comment]
Looks like the best solution right now for me too! Seems to be the same RN-internal issue as in #9252

**Comment by thesahindia** [comment]
The above checkboxes aren't relevant because it was an improvement and technically not a regression

**Comment by thesahindia** [comment]
@arielgreen, it's ready to be paid for the C+ review.

**Comment by arielgreen** [comment]
@thesahindia sent offer in Upwork, please accept when you get a moment.

**Comment by thesahindia** [comment]
Accepted, thanks!

**Comment by arielgreen** [comment]
Excellent, paid.
